### PR TITLE
Removing some unused or unnecessary hack scripts

### PR DESCRIPTION
**Pull request by pseudomuto**
We have a bunch of targets in //hack that check if they're running in bazel and either warn the user to install bazel or subsequently call itself via bazel run/test ....

If we remove the scripts (where possible), we can avoid the need to have these checks, as well as configuring an explicit target in the BUILD files. For most of these, I was able to simply add a new make target and use `bazel run //hack/bin:<cmd> ...`.

### CODE REVIEW

File: hack/BUILD.bazel [hunk 1]
```diff
@@ -129,60 +79,20 @@ sh_test(
...
-    srcs = ["update-gofmt.sh"],
-    args = [
-        "$(location %s)" % GOFMT,
-    ],
-    data = [
-        GOFMT,
-    ],
-)
-
-sh_test(
```
Comment (chrislovecnm) [inline hack/BUILD.bazel:174]: Why are we removing this?

File: pkg/actor/actor.go [hunk 2]
```diff
@@ -18,6 +18,7 @@ package actor
 
 import (
 	"context"
```
Comment (chrislovecnm) [inline pkg/actor/actor.go:20]: Can you remove this please?

Comment (pseudomuto) [reply pkg/actor/actor.go:20]: You mean the new line? The new `make dev/fmt` command added that.

File: hack/BUILD.bazel (hunk 1 above)
Comment (pseudomuto) [reply hack/BUILD.bazel:174]: It's covered with `make dev/lint`

File: hack/BUILD.bazel [hunk 3]
```diff
@@ -129,58 +79,29 @@ sh_test(
...
-    args = [
-        "$(location %s)" % GOFMT,
-    ],
-    data = [
-        GOFMT,
-    ],
-)
-
 sh_test(
     name = "verify-gofmt",
```
Comment (pseudomuto) [inline hack/BUILD.bazel:97]: It's back so that `bazel test //hack/...` runs gofmt

**Review by chrislovecnm — APPROVED** [review]
LGTM

**Pull request merged**